components/ai_chat/resources/page/components/conversation_list/index.tsx

```diff
@@ -14,12 +14,34 @@ import DataContext from '../../state/context'
 import ContextMenuAssistant from '../context_menu_assistant'
 import { getLocale } from '$web-common/locale'
 import SiteTitle from '../site_title'
+import CodeBlock from '../code_block'
```

Comment:
Should we try dynamic import here so that `react-syntax-highlighter` doesn't bloat initial bundle size?

Reply:
we can do dynamic imports but webpack bundles lazy loaded components as raw url to html src element, which is a prohibited behavior as aichat's webui is UntrustedWebUI. i.e `require-trusted-types-for 'script'`

```
[56643:259:1228/085528.893559:ERROR:CONSOLE(42410)] "This document requires 'TrustedScriptURL' assignment.", source: chrome-untrusted://chat/chat_ui.bundle.js (42410)
[56643:259:1228/085528.899552:ERROR:CONSOLE(17363)] "Uncaught TypeError: Failed to set the 'src' property on 'HTMLScriptElement': This document requires 'TrustedScriptURL' assignment.", source: chrome-untrusted://chat/chat_ui.bundle.js (17363)
```

we could tell webpack to output [untrusted](https://webpack.js.org/configuration/output/#outputtrustedtypes) script urls but i believe all WebUIs will have to allow to `createPolicy`? 

for ex, I get an error on ntp webui when changing webpack's output config to spit trusted type urls:

```
[51969:259:1228/083642.617585:ERROR:CONSOLE(71871)] "Refused to create a TrustedTypePolicy named 'brave-core' because it violates the following Content Security Policy directive: "trusted-types default".", source: chrome://newtab/brave_new_tab.bundle.js (71871)
[51969:259:1228/083642.618169:ERROR:CONSOLE(71871)] "Uncaught TypeError: Failed to execute 'createPolicy' on 'TrustedTypePolicyFactory': Policy "brave-core" disallowed.", source: chrome://newtab/brave_new_tab.bundle.js (71871)
```

Reply:
Yes this was one of the benefits that upgrading to webpack 5 brings. Seems fairly simple to get webpack to add a policy named something like "webpack-dynamic-scripts" and allow it for all webui pages from the CSP like we do [right now for the `default` policy](https://github.com/brave/brave-core/blob/master/browser/ui/webui/brave_webui_source.cc#L77-L79).

Please do that in here or an immediate follow-up (which is fine to decouple any problems this may cause other pages). 15% savings is worth it.

Reply:
i can do this as a follow-up, issue: https://github.com/brave/brave-browser/issues/35257

Reply:
overriding the csp to allow default policy and importing defaultTrustedTypesPolicy worked